### PR TITLE
Add dask-cloudprovider recipe

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jacobtomlinson** [comment]
@mrocklin would you like to be added to the maintainers here?

**Comment by mrocklin** [comment]
Sure.  Thanks

**Comment by npavlovikj** [comment]
Does `dask-cloudprovider` only work with Python 2.7? If so, please add `python <3` in both `host` and `run` sections.

**Comment by jacobtomlinson** [comment]
No, its 3.5+. What should I do for that?

**Comment by npavlovikj** [comment]
Ah, then do `python >=3.5` in both sections.

File: recipes/dask-cloudprovider/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - python >=3.5
+    - aiobotocore
+    - dask
+    - distributed
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  imports:
+    - dask-cloudprovider
```
Comment (npavlovikj) [inline recipes/dask-cloudprovider/meta.yaml]: Looks like the import is `dask_cloudprovider`, and not `dask-cloudprovider`.

File: recipes/dask-cloudprovider/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
+  # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
+  noarch: python
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: "{{ PYTHON }} -m pip install . -vv"
+
```
Comment (npavlovikj) [inline recipes/dask-cloudprovider/meta.yaml]: Looks like `dask-ecs` is an executable, so you can an entry point here, something like:
```
  entry_points:
    - dask-ecs=dask_cloudprovider.cli.ecs:go
```

Comment (jacobtomlinson) [reply recipes/dask-cloudprovider/meta.yaml]:
```suggestion

  entry_points:
    - dask-ecs=dask_cloudprovider.cli.ecs:go
```

File: recipes/dask-cloudprovider/meta.yaml (hunk 1 above)
Comment (jacobtomlinson) [reply recipes/dask-cloudprovider/meta.yaml]:
```suggestion
    - dask_cloudprovider
```

File: recipes/dask-cloudprovider/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    - aiobotocore
+    - dask
+    - distributed
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  imports:
+    - dask_cloudprovider
+
```
Comment (npavlovikj) [inline recipes/dask-cloudprovider/meta.yaml]: You should test for `dask-ecs` here as well, something like:
```
commands:
  - dask-ecs -h
```
or whatever will give you exit code 0.

File: recipes/dask-cloudprovider/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+  entry_points:
+    - dask-ecs=dask_cloudprovider.cli.ecs:go
+requirements:
+  build: []
```
Comment (npavlovikj) [inline recipes/dask-cloudprovider/meta.yaml]: Remove `build: []`.

File: recipes/dask-cloudprovider/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (npavlovikj) [inline recipes/dask-cloudprovider/meta.yaml]: Please clean the recipe and remove all the comments.

**Comment by jacobtomlinson** [comment]
@npavlovikj thanks for the review. I've made the requested changes.

**Pull request merged**